presto-hive/src/main/java/com/facebook/presto/hive/metastore/HivePrivilegeInfo.java

```diff
@@ -0,0 +1,102 @@
+ * limitations under the License.
+ */
+package com.facebook.presto.hive.metastore;
+
+import com.facebook.presto.spi.security.Privilege;
+import com.google.common.collect.ImmutableSet;
+import org.apache.hadoop.hive.metastore.api.PrivilegeGrantInfo;
+
+import java.util.Arrays;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static com.facebook.presto.hive.metastore.HivePrivilegeInfo.HivePrivilege.DELETE;
+import static com.facebook.presto.hive.metastore.HivePrivilegeInfo.HivePrivilege.INSERT;
+import static com.facebook.presto.hive.metastore.HivePrivilegeInfo.HivePrivilege.OWNERSHIP;
+import static com.facebook.presto.hive.metastore.HivePrivilegeInfo.HivePrivilege.SELECT;
+import static com.facebook.presto.hive.metastore.HivePrivilegeInfo.HivePrivilege.UPDATE;
+import static java.util.Locale.ENGLISH;
+
+public class HivePrivilegeInfo
+{
+    HivePrivilegeInfo(HivePrivilege hivePrivilege, boolean grantOption)
+    {
+        this.hivePrivilege = hivePrivilege;
+        this.grantOption = grantOption;
+    }
+
+    public HivePrivilege getHivePrivilege()
+    {
+        return hivePrivilege;
+    }
+
+    public void setHivePrivilege(HivePrivilege hivePrivilege)
+    {
+        this.hivePrivilege = hivePrivilege;
+    }
+
+    public boolean isGrantOption()
+    {
+        return grantOption;
+    }
+
+    public void setGrantOption(boolean grantOption)
+    {
+        this.grantOption = grantOption;
+    }
+
+    public enum HivePrivilege
+    {
+        SELECT, INSERT, UPDATE, DELETE, OWNERSHIP;
+    }
+
+    private HivePrivilegeInfo.HivePrivilege hivePrivilege;
+    private boolean grantOption;
+
+    public static Set<HivePrivilegeInfo> parsePrivilege(PrivilegeGrantInfo userGrant)
+    {
+        boolean withGrantOption = userGrant.isGrantOption();
+        String name = userGrant.getPrivilege().toUpperCase(ENGLISH);
+        switch (name) {
+            case "ALL":
+                return Arrays.asList(HivePrivilege.values()).stream()
+                        .map(hivePrivilege -> new HivePrivilegeInfo(hivePrivilege, withGrantOption))
+                        .collect(Collectors.toSet());
+            case "SELECT":
+                return ImmutableSet.of(new HivePrivilegeInfo(SELECT, withGrantOption));
+            case "INSERT":
+                return ImmutableSet.of(new HivePrivilegeInfo(INSERT, withGrantOption));
+            case "UPDATE":
+                return ImmutableSet.of(new HivePrivilegeInfo(UPDATE, withGrantOption));
+            case "DELETE":
+                return ImmutableSet.of(new HivePrivilegeInfo(DELETE, withGrantOption));
+            case "OWNERSHIP":
+                return ImmutableSet.of(new HivePrivilegeInfo(OWNERSHIP, withGrantOption));
+        }
+        return ImmutableSet.of();
+    }
+
+    public static HivePrivilege toHivePrivilege(Privilege privilege)
+    {
+        switch (privilege) {
+            case SELECT:
```

Comment:
I would prefix this with the enum it's coming from. That way it's clear that a conversion is happening and the conversion is from Privilege to HivePrivilege. Otherwise it looks a little strange to convert from SELECT to SELECT.